lib/wda-mac.js

```diff
@@ -282,19 +286,57 @@ class WDAMacServer {
     }
   }
 
+
+  /**
+   * @typedef {Object} ProxyProperties
+   *
+   * @property {string} scheme - The scheme proxy to.
+   * @property {string} host - The host name proxy to.
+   * @property {number} port - The port number proxy to.
+   * @property {string} path - The path proxy to.
+   */
+
+  /**
+   * Returns information about proxy information which WDAMacServer proxy to.
+   *
+   * @param {?string} wdaUrl - The URL for WDAMacServer app to proxy requests to.
+   * @return {ProxyProperties}
+   */
+  parseProxyProperties (wdaUrl) {
+    let scheme = 'http';
+    if (!wdaUrl) {
+      return {scheme, server: DEFAULT_SYSTEM_HOST, port: this.process.port, path: ''};
+    }
+
+    const {protocol, hostname, port, pathname} = url.parse(wdaUrl);
+    if (_.isString(protocol)) {
+      scheme = protocol.split(':')[0];
+    }
+    return {
+      scheme,
+      server: hostname ?? DEFAULT_SYSTEM_HOST,
+      port: _.isNil(port) ? DEFAULT_SYSTEM_PORT : _.parseInt(port),
+      path: pathname === '/' ? '' : pathname
+    };
+  }
+
   async startSession (caps) {
     this.serverStartupTimeoutMs = caps.serverStartupTimeout ?? this.serverStartupTimeoutMs;
-    const wasProcessInitNecessary = await this.process.init(caps);
+    this.isProxyingRemoteWDAMac = !!caps.webDriverAgentMacUrl;
+    const wasProcessInitNecessary = this.isProxyingRemoteWDAMac ? false : await this.process.init(caps);
```

Comment:
Process init should always not be necessary if remote proxy is enabled. We need to patch isProxyReady check instead